### ISSUE TITLE
False positive about nested sequences in Numpy

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#3657]
## Environment data

-   Language Server version: v2022.7.11 Pre-Release
-   OS and version: Win 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.13 Anaconda

## Code Snippet

Here is a simplified version of `numpy._typing._NestedSequence`. This false positive affects a lot of Numpy operations.

```python
from __future__ import annotations

from typing import Protocol, TypeVar, overload

T_co = TypeVar("T_co", covariant=True)


class NestedSequence(Protocol[T_co]):
    @overload
    def __getitem__(self, index: int, /) -> T_co | NestedSequence[T_co]:
        ...

    @overload
    def __getitem__(self, index: slice, /) -> NestedSequence[T_co]:
        ...


a: list[int] = globals()[""]  # e.g. [0, 1, 2]
a_: NestedSequence[int] = a

b: list[int | list[int]] = globals()[""]  # e.g. [0, [1, 2]]
b_: NestedSequence[int] = b

c: list[list[int] | int] = globals()[""]  # e.g. [[0, 1], 2]
c_: NestedSequence[int] = c
```

## Expected behavior

No errors/warnings.

## Actual behavior

![image](https://user-images.githubusercontent.com/15389744/177244508-4e602b9a-5428-47fe-b46e-0bdd86b6edab.png)

## Additional Sample Using Numpy

```python
from __future__ import annotations

import numpy, numpy.typing
# numpy.__version__ == "1.23.0"

a: numpy.typing.NDArray[numpy.float_] = globals()[""]
b: numpy.typing.NDArray[numpy.float_] = a[range(10), 0]  # Legal use but reported.
```

**Comment by erictraut** [comment]
On first inspect, this looks like a false negative rather than a false positive. It looks like the `b_: NestedSequence[int] = b` should also generate an error. This would be consistent with mypy. I'll need to investigate further.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.261, which I just published. It will also be included in an upcoming release of pylance.
